### PR TITLE
[AST] Don’t treat missing arguments as a single () argument

### PR DESCRIPTION
**Pull request by jtbandes**
Hopefully resolves a crashing case of the form `!()`, as found by @practicalswift.

CalleeCandidateInfo was being constructed with 1 argument of type `()`, which resulted in a call to `getElement(0)` even though the argument is actually a TupleType with no elements.

### CODE REVIEW

**Comment by jtbandes** [comment]
@swift-ci please smoke test

**Comment by jtbandes** [comment]
@swift-ci Please smoke test.

@DougGregor, you'll have to decide whether this is actually a *good* fix, but it does seem (miraculously) to fix the crasher without breaking other tests.

**Comment by jtbandes** [comment]
@swift-ci Please test

File: validation-test/compiler_crashers_fixed/28379-swift-constraints-constraintsystem-diagnosefailureforexpr.swift [hunk 1]
```diff
@@ -5,6 +5,6 @@
 // See http://swift.org/LICENSE.txt for license information
 // See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 
-// RUN: not --crash %target-swift-frontend %s -parse
+// RUN: not %target-swift-frontend %s -parse
 // REQUIRES: asserts
```
Comment (jrose-apple) [inline validation-test/compiler_crashers_fixed/28379-swift-constraints-constraintsystem-diagnosefailureforexpr.swift]: Please drop the "REQUIRES: asserts" as well!

**Comment by DougGregor** [comment]
thanks!

**Comment by DougGregor** [comment]
(FWIW, I fixed the REQUIRES: asserts thing before merging)

**Pull request merged**